client/app/vps/backup-storage/vps-backup-storage.controller.js

```diff
@@ -1,70 +1,38 @@
 class VpsBackupStorageCtrl {
-    constructor ($stateParams, $translate, CloudMessage, VpsActionService, VpsService) {
+    constructor ($stateParams, ControllerHelper, VpsActionService, VpsService) {
         this.serviceName = $stateParams.serviceName;
-        this.$translate = $translate;
-        this.CloudMessage = CloudMessage;
+        this.ControllerHelper = ControllerHelper;
         this.VpsActionService = VpsActionService;
         this.VpsService = VpsService;
-
-        this.loaders = {
-            init: false,
-            information: false,
-            vps: false
-        };
-        this.backup = {};
-        this.vps = {};
-
     }
 
-    $onInit () {
-        this.loadInformation();
-        this.loadBackup();
+    initLoaders () {
+        this.backup = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.VpsService.getBackupStorageTab(this.serviceName)
+        });
+        this.info = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.VpsService.getBackupStorageInformation(this.serviceName)
+        });
+        this.vps = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.VpsService.getSelectedVps(this.serviceName)
+        });
     }
 
-    loadVps () {
-        this.loaders.vps = true;
-        this.VpsService.getSelectedVps(this.serviceName)
-            .then(vps => {
-                this.vps = vps;
-            })
-            .catch(err => this.CloudMessage.error(err))
-            .finally(() => { this.loaders.vps = false; });
-    }
+    $onInit () {
+        this.initLoaders();
 
-    loadInformation () {
-        this.loaders.information = true;
-        this.VpsService.getBackupStorageInformation(this.serviceName)
-            .then(backupInfo => {
-                this.backup.information = backupInfo;
-                if (backupInfo.activated === true && backupInfo.quota) {
-                    if (backupInfo.usage === 0) {
-                        backupInfo.usage = {
-                            unit: "%",
-                            value: 0
-                        };
-                    }
-                }
-                if (!backupInfo.activated) {
-                    this.loadVps();
+        this.backup.load();
+        this.info.load()
+            .then(() => {
+                if (!this.info.data.activated) {
```

Comment:
It's a bit hard to know what's inside this.info.data.activated, we don't have a clue why the vps is only loaded if the backup option is deactivated

Reply:
vps is loaded only in that case to know if we can order it or not. Any idea ?